src/lib/analytics/index.ts

```diff
@@ -1,12 +1,29 @@
 import mixpanel from "mixpanel-browser";
 import { env } from "$env/dynamic/public";
 
-const analytics = {
+interface IConfig {
+  [key: string]: any;
+}
+
+interface IProps {
+  [key: string]: any;
+}
+
+interface IAnalytics {
+  init: (config: IConfig) => void;
+  getUserId: () => string;
+  track: {
+    button: (props: IProps) => void;
+    timeOnPage: (props: IProps) => void;
+  };
+}
+
+const analytics: IAnalytics = {
   init: (config) => {
-    mixpanel.init(env.PUBLIC_PROJECT_TOKEN, config);
+    mixpanel.init(env.PUBLIC_PROJECT_TOKEN!, config);
```

Comment:
`!` is this necessary?

Reply:
Typescript me daba problema por estar en modo "strict", ahora la app está funcionando con "strict": false así que ya borré el "!"